src/storage/src/sink/kafka.rs

```diff
@@ -448,27 +421,13 @@ impl KafkaSinkState {
 
         let retry_manager = Arc::new(Mutex::new(KafkaSinkSendRetryManager::new()));
 
-        let (status_tx, mut status_rx) = mpsc::channel(16);
-
         let producer_context = SinkProducerContext {
             metrics: Arc::clone(&metrics),
-            status_tx: status_tx.clone(),
             retry_manager: Arc::clone(&retry_manager),
         };
 
         let healthchecker: Arc<Mutex<Option<Healthchecker>>> = Arc::new(Mutex::new(None));
 
-        {
-            let healthchecker = Arc::clone(&healthchecker);
-            task::spawn(|| "producer-error-report", async move {
-                while let Some(status) = status_rx.recv().await {
-                    if let Some(hc) = healthchecker.lock().await.as_mut() {
-                        hc.update_status(status).await;
-                    }
-                }
-            });
-        }
-
```

Comment:
Is it okay to remove this without the other parts of the other PR? Is there a metadata fetch loop we need to add?

Reply:
yep!